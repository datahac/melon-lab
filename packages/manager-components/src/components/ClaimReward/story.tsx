import { storiesOf } from '@storybook/react';
import React from 'react';
import ClaimReward from './index';

const data = {};

storiesOf('Components|Claim Reward', module).add('Default', () => {
  return <ClaimReward {...data} />;
});
