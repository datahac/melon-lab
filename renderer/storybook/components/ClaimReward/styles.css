.claim-reward__message {
  font-weight: var(--font-weight-bold);
  margin: 0 0 var(--space-m) 0;
}

.claim-reward__steps {
  counter-reset: section;
  margin: 0;
  padding: 0;
  counter-reset: section;
  list-style-type: none;
}

.claim-reward__step {
  font-weight: var(--font-weight-bold);
  padding-top: var(--space-s);
  margin-bottom: var(--space-s);
  border-top: 1px solid var(--color-border);
}

.claim-reward__step-text {
  display: inline-block;
  font-weight: var(--font-weight-regular);
  padding-bottom: var(--space-xs);
}

.claim-reward__step::before {
  counter-increment: section;
  content: 'Step ' counter(section) ': ';
}
