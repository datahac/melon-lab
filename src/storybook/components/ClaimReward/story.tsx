import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import ClaimReward from './index';

const data = {
  competitionName: 'Naxos',
  claimReward: action('claimReward'),
  isCompetitionActive: false,
  endTime: 'Tomorrow',
  redeemParosShares: action('redeemParosShares'),
};

storiesOf('Components|Claim Reward', module).add('Default', () => {
  return <ClaimReward {...data} />;
});
