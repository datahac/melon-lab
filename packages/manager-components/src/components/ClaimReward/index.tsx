import React, { Fragment, StatelessComponent } from 'react';
import Button from '~/blocks/Button';

import styles from './styles.css';

export interface ClaimRewardProps {
  claimReward: () => void;
  competitionName: string;
  endTime?: string;
  isCompetitionActive?: boolean;
  redeemParosShares: () => void;
}

export const ClaimReward: StatelessComponent<ClaimRewardProps> = ({
  claimReward,
  competitionName = 'Naxos',
  endTime,
  isCompetitionActive,
  redeemParosShares,
}) => {
  return (
    <div className="claim-reward">
      <style jsx>{styles}</style>
      <h3>{competitionName} reward</h3>
      {isCompetitionActive ? (
        <div className="claim-reward__message">
          {competitionName} reward can only be claimed at the end of the{' '}
          {competitionName} epoch. The competition will end on {endTime}
        </div>
      ) : (
        <Fragment>
          <div className="claim-reward__message">
            {competitionName} has ended. You can now claim your reward. By
            clicking the "Claim shares" button, the shares of your fund will be
            transfered back to your account. Then click on the "Redeem" button,
            and the underlying assets of your fund will then be transfered to
            your manager address.
          </div>
          <ul className="claim-reward__steps">
            <li className="claim-reward__step">
              <span className="claim-reward__step-text">
                Claim my shares from the {competitionName} contract
              </span>
              <Button onClick={claimReward}>Claim</Button>
            </li>
            <li className="claim-reward__step">
              <span className="claim-reward__step-text">Redeem my shares</span>
              <Button onClick={redeemParosShares}>Redeem</Button>
            </li>
          </ul>
        </Fragment>
      )}
    </div>
  );
};

export default ClaimReward;
