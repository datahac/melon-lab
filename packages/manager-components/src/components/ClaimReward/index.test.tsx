import React from 'react';
import ClaimReward from './index';

const mockCallbackClaimReward = jest.fn();
const mockCallbackRedeemParosShares = jest.fn();
const data = {
  competitionName: 'Naxos',
  claimReward: mockCallbackClaimReward,
  isCompetitionActive: false,
  endTime: 'Tomorrow',
  redeemParosShares: mockCallbackRedeemParosShares,
};

describe('ClaimReward', () => {
  const defaultElement = <ClaimReward {...data} />;
  let customElement;

  it('should render correctly with isCompetitionActive', () => {
    customElement = <ClaimReward {...data} isCompetitionActive={true} />;
    const wrapper = shallow(customElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('onClick claimReward event', () => {
    const wrapper = shallow(defaultElement);
    wrapper
      .find('Button')
      .first()
      .simulate('click');
    expect(mockCallbackClaimReward.mock.calls.length).toBe(1);
  });

  it('onClick redeemParosShares event', () => {
    const wrapper = shallow(defaultElement);
    wrapper
      .find('Button')
      .last()
      .simulate('click');
    expect(mockCallbackRedeemParosShares.mock.calls.length).toBe(1);
  });
});
