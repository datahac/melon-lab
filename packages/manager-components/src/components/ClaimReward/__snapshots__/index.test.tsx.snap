// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimReward should render correctly 1`] = `
<div
  className="jsx-__scopedHash claim-reward"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Naxos
     reward
  </h3>
  <div
    className="jsx-__scopedHash claim-reward__message"
  >
    Naxos
     has ended. You can now claim your reward. By clicking the "Claim shares" button, the shares of your fund will be transfered back to your account. Then click on the "Redeem" button, and the underlying assets of your fund will then be transfered to your manager address.
  </div>
  <ul
    className="jsx-__scopedHash claim-reward__steps"
  >
    <li
      className="jsx-__scopedHash claim-reward__step"
    >
      <span
        className="jsx-__scopedHash claim-reward__step-text"
      >
        Claim my shares from the 
        Naxos
         contract
      </span>
      <Button
        onClick={[MockFunction]}
      >
        Claim
      </Button>
    </li>
    <li
      className="jsx-__scopedHash claim-reward__step"
    >
      <span
        className="jsx-__scopedHash claim-reward__step-text"
      >
        Redeem my shares
      </span>
      <Button
        onClick={[MockFunction]}
      >
        Redeem
      </Button>
    </li>
    <li
      className="jsx-__scopedHash claim-reward__step"
    >
      <span
        className="jsx-__scopedHash claim-reward__step-text"
      >
        Show
         
        <a
          className="jsx-__scopedHash claim-reward__report-link"
          rel="noopener noreferrer"
          target="_blank"
        >
          fund report
        </a>
      </span>
    </li>
  </ul>
</div>
`;

exports[`ClaimReward should render correctly with isCompetitionActive 1`] = `
<div
  className="jsx-__scopedHash claim-reward"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Naxos
     reward
  </h3>
  <div
    className="jsx-__scopedHash claim-reward__message"
  >
    Naxos
     reward can only be claimed at the end of the
     
    Naxos
     epoch. The competition will end on 
    Tomorrow
  </div>
</div>
`;
